Foundation/Unit.swift

```diff
@@ -185,21 +221,45 @@ open class Dimension : Unit {
         self.converter = converter
         super.init(symbol: symbol)
     }
-    
+
+    public required init(symbol: String) {
+        fatalError("You must use the designated initializer.")
```

Comment:
I think this error message will read better as something like `"\(type(of: self) must be initialized with designated initializer \(type(of: self)).init(symbol: String, converter: UnitConverter)"`, because... (further comment below in overrides of this method)

Reply:
I'm afraig I can't simply use `"\(type(of: self)) must be …` though, unless I fully initialize `self`.

So I think our two options are

```swift
public required init(symbol: String) {
    // instance initialized only to provide a more readable error message with class type.
    self.converter = UnitConverter()
    super.init(symbol: symbol)
    fatalError("\(type(of: self)) must be initialized with designated initializer \(type(of: self)).init(symbol: String, converter: UnitConverter)")
}
```
```swift
public required init(symbol: String) {
    fatalError("Dimension subclass must be initialized with designated initializer init(symbol: String, converter: UnitConverter)")
}
```

Which one would you recommend?

Unless there is a third way I'm missing 🤔

Reply:
@bubski The diagnostic here is misleading; this is actually happening because string interpolation actually puts arguments in capturing autoclosures (hence the complaint).

You should be able to check the type up-front:
```swift
let T = type(of: self)
fatalError("\(T) must be initialized with designated initializer \(T).init(symbol: String, converter: UnitConverter)")
```

Reply:
Ha! That's a surprise. Thanks again @itaiferber :)

Reply:
Sure thing! It surprised me too — @phausler just showed this to me, so I'm just passing on the knowledge. 😉